core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -2656,10 +2718,10 @@ public int compareTo(@Nonnull CorDef o) {
    * updated.
    *
    * </ol> */
-  private static class CorelMap {
-    private final Multimap<RelNode, CorRef> mapRefRelToCorRef;
-    private final SortedMap<CorrelationId, RelNode> mapCorToCorRel;
-    private final Map<RexFieldAccess, CorRef> mapFieldAccessToCorRef;
+  protected static class CorelMap {
+    protected final Multimap<RelNode, CorRef> mapRefRelToCorRef;
+    public final SortedMap<CorrelationId, RelNode> mapCorToCorRel;
```

Comment:
this seems "dangerous" to make the set public

Reply:
I changed all members to private and created a getter for mapCorToCorRel. Thanks for noticing this.